packages/xdl/src/project/createBundlesAsync.ts

```diff
@@ -71,18 +113,15 @@ export async function createBundlesAsync(
     }
   }
 
-  const isLegacy = isLegacyImportsEnabled(
-    getConfig(projectRoot, { skipSDKVersionRequirement: true }).exp
-  );
-  // If not legacy, delete the target option to prevent warnings from being thrown.
-  if (!isLegacy) {
-    delete publishOptions.target;
-  }
+  const config = getConfig(projectRoot, { skipSDKVersionRequirement: true });
+  const isLegacy = isLegacyImportsEnabled(config.exp);
   const platforms: Platform[] = ['android', 'ios'];
   const [android, ios] = await bundleAsync(
     projectRoot,
+    publishOptions.target ?? getDefaultTarget(projectRoot),
```

Comment:
```suggestion
```
typo?

Reply:
not really a typo.
this target here is for HermesBundler to determine workflows.
the target below is for [metro-config](https://github.com/expo/expo-cli/blob/a162e4d4166736cbb824b05ea8175bd58a971433/packages/metro-config/src/ExpoMetroConfig.ts#L72-L82) which seems to be removed in the future.

Reply:
ah ok i see.